internal/storage/v1/elasticsearch/factory.go

```diff
@@ -81,12 +82,14 @@ func NewFactoryBase(
 	}
 	f.client.Store(&client)
 
-	if f.config.Authentication.BasicAuthentication.PasswordFilePath != "" {
-		watcher, err := fswatcher.New([]string{f.config.Authentication.BasicAuthentication.PasswordFilePath}, f.onPasswordChange, f.logger)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create watcher for ES client's password: %w", err)
+	if f.config.Authentication.BasicAuthentication.HasValue() {
+		if f.config.Authentication.BasicAuthentication.Get().PasswordFilePath != "" {
```

Comment:
```suggestion
		if file := f.config.Authentication.BasicAuthentication.Get().PasswordFilePath; file != "" {
```
then use `file` variable